src/PreferencesController.ts

```diff
@@ -31,6 +31,10 @@ export class PreferencesController extends BaseController<BaseConfig, Preference
 	 */
 	name = 'PreferencesController';
 
+	private isLocalURL(url: string) {
+		return /:\/\/localhost:/.test(url);
```

Comment:
This fixe the port issue but the we should also be checking for any other non public IP like 127.0.0.1.

```
Mitigation: Simply match against 127.0.0.1, localhost, or any non-public IP address.
```

Reply:
This suggestion from the audit was a little odd; AFAIK, there's no way to know how local hosts are configured on someone's machine. For example, I could locally resolve my local development node to `http://bitpshr.dev:1337` if I set up my host file accordingly. I guess checking readily-known local URLs is better than nothing.

Reply:
I was talking about IPs not hosts.

Reply:
I'm all for adding a cover-all regex for local IPs (candidly I did not know this was a matchable pattern, thanks for this) but as discussed offline, I worry a little about the usefulness of filtering out locally-hosted nodes.

Whether its a developer constantly hitting a local node during development, or a power user constantly hitting a local node for security, it feels like we should save these as "frequent", almost more-so than a public custom RPC network that may've been added by a user to test a one-off dapp.